### PR TITLE
Ling datum antag standarization.

### PR DESCRIPTION
**Pull request by AnturK**
TODO:
- [x] ~Bind ling powers more tightly to the datum, they're not transferable anyway.~ Seems there's mood for ling reworks so i'll skip this one for now.
- [x] Testing.

### CODE REVIEW

**Comment by RemieRichards** [comment]
Do I just abandon my goon ling port then... or?

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -106,8 +106,6 @@
 
 
 /obj/item/clothing/Destroy()
-	if(isliving(loc))
```
Comment (AnturK) [inline code/modules/clothing/clothing.dm]: Before anyone asks, this one is redundant, we deal with this on /item level anyway. If you know of any case where this is needed let me know.

**Comment by AnturK** [comment]
This is mostly repathing tbh, not much actual work here.

**Comment by RemieRichards** [comment]
I'll have a read, see how compatible the two are.

File: code/modules/mob/living/carbon/life.dm [hunk 2]
```diff
@@ -251,12 +251,14 @@
 		if(stat != DEAD)
 			D.stage_act()
 
+//todo generalize this and move hud out
```
Comment (RemieRichards) [inline code/modules/mob/living/carbon/life.dm]: you can leave this as a todo if you want, I've killed it off.

**Comment by RemieRichards** [comment]
Yeah looks fine, I mostly touched (read: gutted/removed) powers, where you've mostly touched the changeling datum.

I partly got powers sharing the Spell system (like the TODO says), so they have action buttons, just a bit fiddly so I may end up throwing out changeling proc holders and just using actions (since with goon ling, the vast majority of powers are already given to the ling, so purchasing and the emporium and stuff are gone already)

File: code/game/gamemodes/changeling/evolution_menu.dm [hunk 3]
```diff
@@ -1,112 +1 @@
...
-		if(hud_used)
-			hud_used.lingstingdisplay.icon_state = null
-			hud_used.lingstingdisplay.invisibility = INVISIBILITY_ABSTRACT
-
-/datum/changeling/proc/has_sting(obj/effect/proc_holder/changeling/power)
-	for(var/obj/effect/proc_holder/changeling/P in purchasedpowers)
-		if(initial(power.name) == P.name)
-			return 1
-	return 0
+//TO BE REMOVED
```
Comment (RemieRichards) [inline code/game/gamemodes/changeling/evolution_menu.dm]: and I did!

**Comment by XDTM** [comment]
Not sure about tying changeling abilities to them being an antag, i feel like they should be managed by a special species datum, or some kind of component if possible.

Especially if antag datums transfer on stuff like mind transference potions.

**Comment by AnturK** [comment]
There's no change from how it works right now, i'm just cleaning it up so i don't have to retrieve antag datum and mind in every proc

**Comment by AnturK** [comment]
Safe to merge. (as safe as it gets without review)

**Comment by Cyberboss** [comment]
Ree

File: code/datums/antagonists/changeling.dm [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+/datum/antagonist/changeling/New()
+	. = ..()
+	generate_name()
+	create_actions()
+
+/datum/antagonist/changeling/Destroy()
+	qdel(cellular_emporium)
+	cellular_emporium = null
+	qdel(emporium_action)
+	emporium_action = null
```
Comment (Cyberboss) [inline code/datums/antagonists/changeling.dm]: QDEL_NULL macro

**Comment by Cyberboss** [comment]
>Asking me to understand gamemode/antag code

**Comment by RemieRichards** [comment]
>3 days ago

#woops 
#weekendcodingtime

**Pull request merged**